modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/WhitelistLoader.java

```diff
@@ -178,41 +184,125 @@ public static Whitelist loadFromResourceFiles(Class<?> resource, String... filep
                             throw new IllegalArgumentException("invalid class definition: failed to parse class name [" + line + "]");
                         }
 
+                        parseType = "class";
                         whitelistClassOrigin = "[" + filepath + "]:[" + number + "]";
                         javaClassName = tokens[0];
 
                         // Reset all the constructors, methods, and fields to support a new class.
                         whitelistConstructors = new ArrayList<>();
                         whitelistMethods = new ArrayList<>();
                         whitelistFields = new ArrayList<>();
+                    } else if (line.startsWith("static ")) {
+                        // Ensure the final token of the line is '{'.
+                        if (line.endsWith("{") == false) {
+                            throw new IllegalArgumentException(
+                                    "invalid class definition: failed to parse class opening bracket [" + line + "]");
```

Comment:
Should the message here indicate "static" opening bracket, not class?

Reply:
Fixed.